shim.js

```diff
@@ -16,13 +16,10 @@ if (searchParams.get("shim") !== "0") {
     const target = document.querySelector(location.hash);
     openAncestorDetailsRecursively(target);
   });
-  // Open ancestor details recursively when <a> element with a fragment is clicked
-  document.querySelectorAll("a").forEach((a) => {
-    a.addEventListener("click", () => {
-      const href = a.getAttribute("href");
-      if (!href || !href.startsWith("#")) return;
-      const target = document.querySelector(href);
-      openAncestorDetailsRecursively(target);
-    });
+  // Open ancestor details recursively when the fragment is changed
+  window.addEventListener('hashchange', () => {
+    if (!location.hash.startsWith("#")) return;
+    const target = document.querySelector(location.hash);
+    openAncestorDetailsRecursively(target);
```

Comment:
Hmmm... It seems that scrolling by fragment navigation is complete when the hashchange event is fired.

So even though the details element is expanded in the hashchange event listener, the browser does not scroll to the details element.

https://github.com/user-attachments/assets/8e1a5af4-104e-4fcb-be39-8fcb4562cf87

Reply:
If we call `target.scrollIntoView()` after expanding the details element, the browser can scroll to the element. However, the scrolling will occur after the hashchange event. The behavior differs from browsers implementing the [Auto-expand details elements](https://chromestatus.com/feature/5032469667512320) feature.

I think that deviations from the standard specification should be avoided.